src/js_types.hpp

```diff
@@ -351,11 +351,12 @@ REALM_JS_INLINE void set_internal(const typename T::Object &object, typename Cla
 
 template<typename T>
 inline bool Value<T>::is_valid_for_property(ContextType context, const ValueType &value, const Property& prop)
-{
-    if (prop.is_nullable && (is_null(context, value) || is_undefined(context, value))) {
+{ 
+    if (is_nullable(prop.type) && (is_null(context, value) || is_undefined(context, value))) {
         return true;
     }
 
+    //lubo
     using PropertyType = realm::PropertyType;
     switch (prop.type) {
```

Comment:
also thanks for picking up this for review. I was doing the changes based on the changes in object store. I noticed the is_array change here https://github.com/realm/realm-object-store/commit/22420458794e0eabf032fd0e406e28fd06157f41#diff-99a0771b4ad4157fdbcf45dfdc8f1e24R151

but PropertyType: Object is left inside the switch statement.